### DISCUSSION TITLE
Unicode characters that have 2 or more Chars

### DISCUSSION THREAD

**Discussion by mio-19** [lihaoyi/fastparse#316] [General]
How do fastparse interact with characters that can't be represented by a `Char`? What are recommended practices for dealing with such characters?

**Comment by lihaoyi** [comment]
It parses them one Char at a time. Typically that works fine

**Reply by mio-19** [reply]
CharsWhile and other api don't work if I want to use a predicate on such characters for example java.lang.Character.isLetter(int codePoint)

**Reply by lihaoyi** [reply]
That's a good point. I don't think we've ever encountered that before. I guess it's unsupported until someone comes up with some proposal for how that should be handled; maybe need some kind of `UnicodeCharsWhile` macro smart enough to iterate over unicode code points and providing an `Int` to the predicate

**Reply by He-Pin** [reply]
@mio-19 Are u parsing jsonpath?

**Reply by mio-19** [reply]
> jsonpath

I am parsing a programming language that supports unicode.
